src/autogluon/cloud/cluster/ray_aws_cluster_config_generator.py

```diff
@@ -0,0 +1,146 @@
+import os
+from typing import Any, Dict, Optional, Union
+
+from .cluster_config_generator import DEFAULT_CONFIG_LOCATION
+from .constants import (
+    AVAILABLE_NODE_TYPES,
+    BLOCK_DEVICE_MAPPINGS,
+    DOCKER,
+    EBS,
+    IMAGE,
+    INSTANCE_TYPE,
+    MAX_WORKERS,
+    MIN_WORKERS,
+    NODE_CONFIG,
+    PROVIDER,
+    REGION,
+    VOLUMN_SIZE,
+)
+from .ray_cluster_config_generator import RayClusterConfigGenerator
+
+
+class RayAWSClusterConfigGenerator(RayClusterConfigGenerator):
+    default_config_file = os.path.join(DEFAULT_CONFIG_LOCATION, "ray_aws_default_cluster_config.yaml")
+
+    def __init__(
+        self, config: Optional[Union[str, Dict[str, Any]]] = None, region: Optional[str] = "us-east-1", **kwargs
+    ) -> None:
+        """
+        Parameter
+        ---------
+        config, Optional[Union[str, Dict[str, Any]]]
+            Config to be used to launch up the cluster. Default: None
+            If not set, will use the default config pre-defined.
+            If str, must be a path pointing to a yaml file containing the config.
+        region, Optional[str]
+            Region to launch the cluster. Default us-east-1
+        """
+        super().__init__(config=config)
+        self._update_region(region)
+
+    def _update_config(
+        self,
+        instance_type: Optional[str] = None,
+        instance_count: Optional[str] = None,
```

Comment:
Why not set it to 1 as a default and change type to `int` in the API?
`instance_count: int = 1` ?

Reply:
It should be int, that's a mistake. Though if we specify a default value for instance_count, we will always update the config to have instance_count = 1 even when user left it blank. I'll stick with default = None, meaning user doesn't want to update it